src/modules/job-post/job-post.service.ts

```diff
@@ -0,0 +1,81 @@
+import { HttpException, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { JobPost } from './entities/job-post.entity';
+import { CreateJobPostDto } from './dto/create-job-post.dto';
+import { Repository } from 'typeorm';
+import { UpdateJobPostDto } from './dto/update-job-post.dto';
+
+@Injectable()
+export class JobPostService {
+  constructor(
+    @InjectRepository(JobPost)
+    private readonly jobPostRepository: Repository<JobPost>,
+  ) {}
+
+  async create(createJobPostDto: CreateJobPostDto) {
+    return await this.jobPostRepository.save({
+      title: createJobPostDto.title,
+      duration: createJobPostDto.duration,
+      durationType: createJobPostDto.durationType,
+      jobDescription: createJobPostDto.jobDescription,
+      rate: createJobPostDto.rate,
+      englishLevel: createJobPostDto.englishLevel,
+      skills: createJobPostDto.skillsId.map((value) => ({
+        id: value,
+      })),
+      category: createJobPostDto.categoryId,
+      user: { id: createJobPostDto.userId },
+    });
```

Comment:
Can't you use  
```typescript
return await this.jobPostRepository.save({
      ...createJobPostDto
    });
```
instead of typing every field in the object?

Reply:
spread is not an option in this case I believe.

Reply:
What about this? Just highlighting the fields that need special treatment to be created
```typescript
async create(createJobPostDto: CreateJobPostDto) {
    return await this.jobPostRepository.save({
      ...createJobPostDto,
      skills: createJobPostDto.skillsId.map((value) => ({
        id: value,
      })),
      user: { id: createJobPostDto.userId },
    });
  }
```